docs/user_manual/performance-guide.md

```diff
@@ -6,6 +6,57 @@ SPDX-License-Identifier: MPL-2.0
 
 # Guidelines for performance enhancement
 
+The `power-grid-model` is a library that shines in its ability to handle calculations at scale.
+It remains performant, even when doing calculations with one or a combination of the following extremes (non-exhaustive):
+
+- Large grids
+- Batch calculations with many scenarios
+- Many changes in the grid in each scenario
+
+To achieve that high performance, several optimizations to the code are made.
+To use those optimizations to their fullest extend, we recommend our users to follow the following guidelines.
+
+## Data validity
+
+Many of our optimizations rely on assuming input data validity and the fact that the provided grid is reasonably close to realistic.
+Non-convergence, underdetermined equations (sparse matrices) or other unexpected behavior may therefore be encountered when the data is not realistic.
```

Comment:
What is the reason that you mention `sparse matrices` here specifically under `underdetermined equations`? It's unclear how these two are necessarily related.

Reply:
I think I agree, because even with a well determined set of equations, the resulting matrices are very sparse, right? And that is even the reason why have our own math solvers.

Reply:
the reason is specifically about the `SparseMatrixError` that's dreaded by many users. This was an attempt to elaborate that. Do you know of a better way to explain that? or should i just remove it entirely?